src/main/java/hudson/os/windows/WindowsRemoteLauncher.java

```diff
@@ -47,15 +47,19 @@ private String buildCommandLine(ProcStarter ps) {
 
     @Override
     public Proc launch(ProcStarter ps) throws IOException {
-        maskedPrintCommandLine(ps.cmds(), ps.masks(), ps.pwd());
+        FilePath pwd = ps.pwd();
+        if (pwd == null) {
+            throw new IOException();
```

Comment:
Maybe with a message would help clarify if this happens?

Reply:
I try to avoid exceptions without message (not always possible), so I agree about adding the message. In this case, knowing why we couldn't get the current directory would be really useful.

Reply:
I agree, I just need to build a good message..